File: Answer/1848.到目标元素的最小距离.java
/*
 * @lc app=leetcode.cn id=1848 lang=java
 *
 * [1848] 到目标元素的最小距离
 */

// @lc code=start
class Solution {
    public int getMinDistance(int[] nums, int target, int start) {
        int n = nums.length;
        int left = start, right = start;
        
        // 向两边扩散查找
        while (left >= 0 || right < n) {
            if (left >= 0 && nums[left] == target) {
                return start - left;
            }
            if (right < n && nums[right] == target) {
                return right - start;
            }
            left--;
            right++;
        }
        
        return -1;
    }
}
// @lc code=end

File: Answer/1848.到目标元素的最小距离.go
/*
 * @lc app=leetcode.cn id=1848 lang=golang
 *
 * [1848] 到目标元素的最小距离
 */

// @lc code=start
func getMinDistance(nums []int, target int, start int) int {
	n := len(nums)
	ans := -1
	for i := 0; i < n; i++ {
		if start+i < n && nums[start+i] == target {
			ans = i
			break
		}
		if start-i >= 0 && nums[start-i] == target {
			ans = i
			break
		}
		if start+i >= n && start-i < 0 {
			break
		}
	}
	return ans
}

// @lc code=end
